ldclient/twisted.py

```diff
@@ -0,0 +1,179 @@
+                    log.warning('ProtocolError exception caught while getting flag. Retrying.')
+                    d = yield run(False)
+                    defer.returnValue(d)
+                else:
+                    log.exception('Unhandled exception. Returning default value for flag.')
+                    defer.returnValue(default)
+            except Exception:
+                log.exception('Unhandled exception. Returning default value for flag.')
+                defer.returnValue(default)
+
+        return run(True)
+
+    @defer.inlineCallbacks
+    def _toggle(self, key, user, default):
+        if self._config._stream and self._stream_processor.initialized():
+            feature = self._stream_processor.get_feature(key)
+        else:
+            hdrs = _headers(self._api_key)
+            uri = self._config._base_uri + '/api/eval/features/' + key
+            r = yield self._session.get(uri, headers=hdrs, timeout=(self._config._connect, self._config._read))
+            r.raise_for_status()
+            feature = r.json()
+        val = _evaluate(feature, user)
+        if val is None:
+            val = default
+        defer.returnValue(val)
+
+
+class TwistedConfig(Config):
+    def __init__(self, *args, **kwargs):
+        super(TwistedConfig, self).__init__(*args, **kwargs)
+        self._stream_processor_class = TwistedStreamProcessor
+
+
+class TwistedStreamProcessor(object):
+
+    def __init__(self, api_key, config):
+        self._store = config._feature_store_class()
+        self.sse_client = TwistedSSEClient(config._stream_uri + "/", headers=_stream_headers(api_key),
+                                           verify=config._verify,
+                                           on_event=partial(StreamProcessor.process_message, self._store))
+        self.running = False
+
+    def start(self):
+        self.sse_client.start()
+        self.running = True
+
+    def stop(self):
+        self.sse_client.stop()
+
+    def get_feature(self, key):
+        return self._store.get(key)
+
+    def initialized(self):
+        return self._store.initialized()
+
+    def is_alive(self):
+        return self.running
+
+
+class TwistedConsumer(object):
+    def __init__(self, session, queue, api_key, config):
+        self._queue = queue
+        """ @type: queue.Queue """
+        self._session = session
+        """ :type: txrequests.Session """
+
+        self._api_key = api_key
+        self._config = config
+        self._flushed = None
+        """ :type: Deferred """
+        self._looping_call = None
+        """ :type: LoopingCall"""
+
+    def start(self):
+        self._flushed = defer.Deferred()
+        self._looping_call = task.LoopingCall(self._consume)
+        self._looping_call.start(5)
+
+    def stop(self):
+        self._looping_call.stop()
+
+    def is_alive(self):
+        return self._looping_call is not None and self._looping_call.running
+
+    def flush(self):
```

Comment:
If flush is called on an empty queue, it will block until the next time `_consume` has actual work to do. This could be longer than 5 seconds if the queue is empty. 

I also think `flush` should force an immediate request with the current queue contents, rather than merely acting as a synchronization point indicating that the next `_consume` cycle has completed.